File: src/main/java/com/radixpro/enigma/shared/common/SessionState.java
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */

package com.radixpro.enigma.shared.common;

import com.radixpro.enigma.ui.domain.FullChart;
import com.radixpro.enigma.xchg.domain.config.Configuration;
import org.apache.log4j.Logger;

/**
 * Remembers state of several components.
 * Implemented as singleton. Is mutable.
 */
public class SessionState {

   private static final Logger LOG = Logger.getLogger(SessionState.class);
   private static SessionState instance = null;
   private FullChart selectedChart;
   private Configuration selectedConfig;

   private SessionState() {
      // prevent instantiation
   }

   public static SessionState getInstance() {
      if (instance == null) {
         instance = new SessionState();
         LOG.info("Created singleton instance for ChartsSessionState.");
      }
      return instance;
   }

   public void deSelectChart() {
      selectedChart = null;
   }

   public FullChart getSelectedChart() {
      return selectedChart;
   }

   public void setSelectedChart(FullChart selectedChart) {
      this.selectedChart = selectedChart;
   }

   public boolean selectedChartIsSet() {
      return selectedChart != null;
   }

   public Configuration getSelectedConfig() {
      return selectedConfig;
   }

   public void setSelectedConfig(Configuration selectedConfig) {
      this.selectedConfig = selectedConfig;
   }

   public boolean selectedConfigIsSet() {
      return selectedConfig != null;
   }


}

File: src/main/java/com/radixpro/enigma/SessionState.kt
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */
package com.radixpro.enigma

import com.radixpro.enigma.domain.astronpos.FullChart
import com.radixpro.enigma.domain.config.Configuration

/**
 * Remembers state of several components.
 * Implemented as singleton. Is mutable.
 */
object SessionState {
    var selectedChart: FullChart? = null
    var selectedConfig: Configuration? = null
    fun deSelectChart() {
        selectedChart = null
    }

    fun selectedChartIsSet(): Boolean {
        return selectedChart != null
    }

    fun selectedConfigIsSet(): Boolean {
        return selectedConfig != null
    }

}